indico/modules/events/schemas.py

```diff
@@ -5,14 +5,56 @@
 # modify it under the terms of the MIT License; see the
 # LICENSE file for more details.
 
-from marshmallow import fields
+import re
+
+from marshmallow import ValidationError, fields, validates_schema
+from marshmallow.validate import Length
 
 from indico.core.marshmallow import mm
 from indico.modules.events.models.events import Event
 from indico.modules.events.models.principals import EventPrincipal
+from indico.util.i18n import _
 from indico.util.marshmallow import PrincipalPermissionList
 
 
+class ValidRegExField(fields.Field):
+    """A valid Python regular expression."""
+
+    def _serialize(self, value, attr, obj, **kwargs):
+        return value.pattern
+
+    def _deserialize(self, value, attr, data, **kwargs):
+        if len(value) < 2:
+            raise ValidationError(_('Empty regex'))
+        try:
+            return re.compile(value)
+        except ValueError as exc:
+            raise ValidationError(str(exc))
+
+
+class AutoLinkerRuleSchema(mm.Schema):
+    """A regex -> url with placeholders mapping rule."""
+
+    regex = ValidRegExField()
+    url = fields.String(validate=Length(min=1))
+
+    @validates_schema
+    def _check_url_is_well_formed(self, data, **kwargs):
+        regex = data['regex']
+        if not regex:
+            raise ValidationError(_('No regex provided'))
+
+        # check if all groups are used
+        for n in range(1, regex.groups + 1):
```

Comment:
hm, named groups vs positional groups... personally i'd go for named groups but I guess either way is fine (i'm just used to named groups from the mattermost autolinker ;))

Reply:
I don't have a strong opinion, but aren't named groups a bit of an overkill here?

Reply:
depending on the pattern it's easier to read. but yeah, i wouldn't bother changing it...